src/main/java/ca/on/oicr/pineryreports/reports/impl/StockReport.java

```diff
@@ -0,0 +1,183 @@
+package ca.on.oicr.pineryreports.reports.impl;
+
+import static ca.on.oicr.pineryreports.util.SampleUtils.*;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.ParseException;
+
+import ca.on.oicr.pinery.client.HttpResponseException;
+import ca.on.oicr.pinery.client.PineryClient;
+import ca.on.oicr.pineryreports.data.ColumnDefinition;
+import ca.on.oicr.pineryreports.reports.TableReport;
+import ca.on.oicr.ws.dto.SampleDto;
+
+import com.google.common.collect.Sets;
+import com.itextpdf.layout.property.TextAlignment;
+
+/**
+ * PCSI Report Request https://jira.oicr.on.ca/browse/GLT-1892
+ */
+public class StockReport extends TableReport {
+  
+  public static final String REPORT_NAME = "stock";
+
+  private final List<ColumnDefinition> columns = Collections.unmodifiableList(Arrays.asList(
+      new ColumnDefinition("Stock Created"),
+      new ColumnDefinition("MISO Alias"),
+      new ColumnDefinition("External Name"),
+      new ColumnDefinition("Conc. (ng/µl)", 50f, TextAlignment.RIGHT),
+      new ColumnDefinition("Vol. (µl)", 50f, TextAlignment.RIGHT),
+      new ColumnDefinition("Total Yield (ng)", 70f, TextAlignment.RIGHT),
+      new ColumnDefinition("Date Received", TextAlignment.CENTER),
+      new ColumnDefinition("Institution")
+  ));
+  
+  private static final String dateRegex = "\\d{4}-\\d{2}-\\d{2}";
+  
+  private String project;
+  private String start;
+  private String end;
+  
+  private List<SampleDto> stocks;
+  private Map<String, SampleDto> allSamplesById;
+  
+  @Override
+  public String getReportName() {
+    return REPORT_NAME;
+  }
+
+  @Override
+  public Collection<Option> getOptions() {
+    Set<Option> opts = Sets.newHashSet();
+    
+    opts.add(Option.builder()
+        .longOpt("project")
+        .hasArg()
+        .argName("code")
+        .required()
+        .desc("Project to report on (required)")
+        .build());
+    opts.add(Option.builder()
+        .longOpt("after")
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "after" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)